pkg/resources/envoy/configmap.go

```diff
@@ -64,6 +64,13 @@ func GenerateEnvoyConfig(kc *v1beta1.KafkaCluster, log logr.Logger) string {
 	var clusters []*envoyapi.Cluster
 
 	for _, brokerId := range util.GetBrokerIdsFromStatus(kc.Status.BrokersState, log) {
+		brokerConfigGroup := util.GetBrokerConfigGroupFromStatus(kc.Status.BrokersState, brokerId, log)
+		if envoyConfig.EnvoyPerBrokerGroup && brokerConfigGroup != "" && brokerConfigGroup != envoyConfig.Id {
+			// Since `EnvoyPerBrokerGroup` is enabled, we add only brokers having the same group as the envoy.
+			// If we cannot retrieve the broker's configGroup from the Status, we will add the broker to all envoys
+			// (this is a safe net to ensure that the brokers are always reachable through envoy)
+			continue
+		}
```

Comment:
I would move the GetBrokerConfigGroupFromStatus evaluation under `envoyConfig.EnvoyPerBrokerGroup` check
```suggestion
		if envoyConfig.EnvoyPerBrokerGroup {
			brokerConfigGroup := util.GetBrokerConfigGroupFromStatus(kc.Status.BrokersState, brokerId, log)
			if brokerConfigGroup != "" && brokerConfigGroup != envoyConfig.Id {
				// Since `EnvoyPerBrokerGroup` is enabled, we add only brokers having the same group as the envoy.
				// If we cannot retrieve the broker's configGroup from the Status, we will add the broker to all envoys
				// (this is a safe net to ensure that the brokers are always reachable through envoy)
				continue
			}
		}
```